lib/europeana/proxy/media.rb

```diff
@@ -96,7 +96,11 @@ def proxy?(env)
       # @return [Hash] rewritten request env
       def rewrite_env(env)
         env['app.record_id'] = env['REQUEST_PATH']
-        edm = Europeana::API.record(env['app.record_id'])['object']
+        env['app.permitted_api_urls'] = ENV['ALLOWED_API_URLS'].split(',').map(&:strip) if ENV['ALLOWED_API_URLS']
```

Comment:
There is no need to do this per-request.

`#initialize` only gets called once when the Rack app is bootstrapped, then `#call` for each request, so set an instance variable like `@permitted_ai_urls` in `#initialize', and go with either "permitted" or "allowed" in both that instance var name and the env var name.